tests/integration/src/test/java/org/apache/pulsar/tests/integration/schema/SchemaTest.java

```diff
@@ -190,32 +186,77 @@ public void testAvroLogicalType() throws Exception {
-            assertEquals(messageForSend.getTimeMicros(), received.getTimeMicros());
-            assertEquals(messageForSend.getTimeMillis(), received.getTimeMillis());
-            assertEquals(messageForSend.getTimestampMicros(), received.getTimestampMicros());
-            assertEquals(messageForSend.getTimestampMillis(), received.getTimestampMillis());
-            assertEquals(messageForSend.getDate(), received.getDate());
-
-            log.info("Successfully consumer avro logical type message : {}", received);
-        }
+                .subscriptionName("test")
+                .subscribe();
+
+        producer.send(messageForSend);
+        log.info("Successfully published avro logical type message : {}", messageForSend);
 
+        AvroLogicalType received = consumer.receive().getValue();
+        assertEquals(messageForSend.getDecimal(), received.getDecimal());
+        assertEquals(messageForSend.getTimeMicros(), received.getTimeMicros());
+        assertEquals(messageForSend.getTimeMillis(), received.getTimeMillis());
+        assertEquals(messageForSend.getTimestampMicros(), received.getTimestampMicros());
+        assertEquals(messageForSend.getTimestampMillis(), received.getTimestampMillis());
+        assertEquals(messageForSend.getDate(), received.getDate());
 
+        log.info("Successfully consumer avro logical type message : {}", received);
+    }
+
+    @Test
+    public void testAutoConsumeSchemaSubscribeFirst() throws Exception {
+        final String tenant = PUBLIC_TENANT;
+        final String namespace = "test-namespace-" + randomName(16);
+        final String topic = "test-auto-consume-schema";
+        final String fqtn = TopicName.get(
+                TopicDomain.persistent.value(),
+                tenant,
+                namespace,
+                topic
+        ).toString();
+
+        admin.namespaces().createNamespace(
+                tenant + "/" + namespace,
+                Sets.newHashSet(pulsarCluster.getClusterName())
+        );
+
+        Consumer<GenericRecord> consumer = client
+                .newConsumer(Schema.AUTO_CONSUME())
+                .topic(fqtn)
+                .subscriptionName("test")
+                .subscribe();
+
+        new Thread(() -> {
+
+            GenericRecord genericRecord = null;
+            try {
+                genericRecord = consumer.receive().getValue();
+            } catch (PulsarClientException e) {
+                e.printStackTrace();
+            }
+            assertEquals(genericRecord.getField("name"), "Tom Hanks");
+            assertEquals(genericRecord.getField("age"), 60);
+            Thread.currentThread().interrupt();
+        }).start();
+
+        Producer<Person> producer = client
+                .newProducer(Schema.AVRO(Person.class))
+                .topic(fqtn)
+                .create();
+
+        Person person = new Person();
+        person.setName("Tom Hanks");
+        person.setAge(60);
+        producer.send(person);
```

Comment:
the test here is inaccurate because you don't wait the thread to complete.

but in fact, I don't think you need to start a separated thread to do so.

You can just do the followings:

```
Producer<Person> producer = client
                .newProducer(Schema.AVRO(Person.class))
                .topic(fqtn)
                .create();

         Person person = new Person();
        person.setName("Tom Hanks");
        person.setAge(60);
        producer.send(person);

GenericRecord genericRecord = null;
            try {
                genericRecord = consumer.receive().getValue();
            } catch (PulsarClientException e) {
                e.printStackTrace();
            }
            assertEquals(genericRecord.getField("name"), "Tom Hanks");
            assertEquals(genericRecord.getField("age"), 60);

```